src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCache.cs

```diff
@@ -344,6 +346,7 @@ public TimeSpan PollingInterval
         private MemoryCache()
         {
             _name = "Default";
+            _disposedName = typeof(MemoryCache).FullName + $"({_name})";
             Init(null);
```

Comment:
You already have the `_name` field so you can construct this string at any time. I suggest creating a `ThrowIfDisposed` method which checks the `IsDisposed` property and the `_throwOnDisposed` field, then constructs the name there. It's only ever used when throwing, so no need to keep it around for the life of the object when it's generatable.  

Also this type isn't sealed, should you be using `this.GetType().FullName` in case someone has derived MemoryCache?